azure-pipelines.yml

```diff
@@ -32,30 +29,33 @@ stages:
           - script: |
               call "%ProgramFiles%\Microsoft Visual Studio\2022\Preview\Common7\Tools\VsDevCmd.bat" ^
                 -host_arch=amd64 -arch=amd64 -no_logo
-              cmake -G Ninja -S $(Build.SourcesDirectory)/tools/format -B $(tmpDir)/format-build
-              cmake --build $(tmpDir)/format-build
-            displayName: 'clang-format'
+              cmake -G Ninja -S $(Build.SourcesDirectory)/tools -B $(tmpDir)/format-validate-build
+              cmake --build $(tmpDir)/format-validate-build
+            displayName: 'Build format and validation'
             timeoutInMinutes: 5
             condition: succeededOrFailed()

```

Comment:
Originally (also before https://github.com/microsoft/STL/pull/2671 ), `'clang-format'` and `'Validate Files'` were marked `condition: succeededOrFailed()`. I think we were just confused about `'clang-format'` (that step runs first), but for Validate Files it was important - this meant that if a PR would fail both the Format and Validate steps, they would see both errors (instead of seeing only the Format problems, fixing those, and then finding non-ASCII characters in Validate or whatever).

I believe that 2 further changes should be made here:

1. Remove `condition: succeededOrFailed()` from `'Build format and validation'`.
   * If something has failed this early like `checkout`, we should just stop.
2. Add `condition: succeededOrFailed()` to `'Validate Files'`.
   * Even when `run-format` fails, we'll still `run-validate`.